src/aioswitcher/api/__init__.py

```diff
@@ -162,20 +171,33 @@ async def _login(self) -> Tuple[str, SwitcherLoginResponse]:
 
         """
         timestamp = current_timestamp_to_hexadecimal()
-        if (
+        if self._token != "":
+            packet = packets.LOGIN_TOKEN_PACKET_TYPE2.format(
+                self._token, timestamp, self._device_id
+            )
+        elif (
             self._device_type
             and self._device_type == DeviceType.BREEZE
             or self._device_type == DeviceType.RUNNER
             or self._device_type == DeviceType.RUNNER_MINI
         ):
-            packet = packets.LOGIN2_PACKET_TYPE2.format(timestamp, self._device_id)
+            packet = packets.LOGIN_PACKET_TYPE2.format(timestamp, self._device_id)
         else:
             packet = packets.LOGIN_PACKET_TYPE1.format(timestamp, self._device_key)
         signed_packet = sign_packet_with_crc_key(packet)
 
         logger.debug("sending a login packet")
         self._writer.write(unhexlify(signed_packet))
         response = await self._reader.read(1024)
+
+        if self._token != "":
+            packet = packets.LOGIN2_TOKEN_PACKET_TYPE2.format(
+                self._device_id, timestamp, self._token
+            )
+            signed_packet = sign_packet_with_crc_key(packet)
+            logger.debug("sending a login2 packet")
+            self._writer.write(unhexlify(signed_packet))
+            response = await self._reader.read(1024)
```

Comment:
1. Why not bool(self._token)?
2. Should this section be merged with the section in row 174 instead of evaluating for the same expression twice?

Reply:
1. It is done
2. No because this is a different packet sent and this should come after. so basically there are 2 calls one after the other only if bool(self._token)

Reply:
@TomerFi Can it be resolved?